core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1291,11 +1291,11 @@ abstract class RDD[T: ClassTag](
     } else {
       val buf = new ArrayBuffer[T]
       val totalParts = this.partitions.length
-      var partsScanned = 0
+      var partsScanned = 0L
```

Comment:
Ah, you're right. `partScanned` cannot exceed the value of `totalParts`.
I'll return it to `Int`.